### ISSUE TITLE
Host device assignment through k8s device plugins

### ISSUE THREAD

**Issue by bc185174** [kubevirt/kubevirt#12086]
**Is your feature request related to a problem? Please describe**:

Host device assignment is quite the manual process. Administrators have to grab the PID/VID which is not guaranteed to be unique and set this as configuration in the KubeVirt CR.

**Describe the solution you'd like**:

- Host device assignment through Kubernetes device plugins https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/.
- Configuration of host permitted devices can be set using regex on device paths and/or device plugins.

**Describe alternatives you've considered**:

N/A

**Additional context**:

**Comment by jean-edouard** [comment]
@vladikr PTAL

**Comment by victortoso** [comment]
> Configuration of host permitted devices can be set using regex on device paths and/or device plugins.

Can you give an actual example of how that would work?

**Comment by alicefr** [comment]
@bc185174 one thing you can always do is to develop your own device plugin and use it with kubevirt. If the env variable are set correctly, then it should work.

**Comment by victortoso** [comment]
> Bus and device number will not work - restarting the node or removing/adding a device will cause the bus/device numbers to change.

Yes, uniqueness of USB devices are complicated. At a single point in time, only one BUS/DEVNo. will exist while we could have multiple vendor/product at any time. We should be able to add more filter-like information but you see is not trivial.

Any suggestions?
